core/src/test/java/org/apache/calcite/rex/RexNormalizeTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.rex;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+/** Test cases for {@link RexNormalize}. */
+class RexNormalizeTest extends RexProgramTestBase {
+
+  @Test void digestIsNormalized() {
+    final RexNode node0 = and(or(vBool(1), vBool(0)), vBool(0));
+    final RexNode node1 = and(vBool(0), or(vBool(0), vBool(1)));
+    assertNodeEquals(node0, node1, true);
```

Comment:
It looks like the nodes are used only once, so I would suggest to inline `node0`, `node1` and similar variables.

For instance:

```java
assertNodeEquals(
    and(or(vBool(1), vBool(0)), vBool(0)),
    and(vBool(0), or(vBool(0), vBool(1)))
);
```

Reply:
An alternative option is to use `Parameterized` tests.

Reply:
Will change to inline, thanks ~